docs/configuration-reference/components/azure-arc-onboarding.md

```diff
@@ -0,0 +1,119 @@
+services](https://azure.microsoft.com/en-in/services/azure-arc/hybrid-data-services/).
+
+- Establish central visibility in the [Azure portal](https://azure.microsoft.com/en-in/features/azure-portal/)
+and enable multi-environment search with Azure Resource Graph.
+
+- Meet [governance](https://azure.microsoft.com/en-in/solutions/governance/) and compliance standards for
+apps, infrastructure and data with [Azure Policy](https://azure.microsoft.com/en-in/services/azure-policy/).
+
+- Delegate access and manage security policies for resources using role-based access control (RBAC) and [Azure
+Lighthouse](https://azure.microsoft.com/en-in/services/azure-lighthouse/).
+
+- Organise and inventory assets through a variety of Azure scopes, such as management groups, subscriptions,
+resource groups and tags.
+
+This component onboards or removes a Lokomotive cluster with Azure Arc.
+
+## Prerequisites
+
+1. Register providers for Azure Arc enabled Kubernetes:
+   ```bash
+   az provider register --namespace Microsoft.Kubernetes
+   az provider register --namespace Microsoft.KubernetesConfiguration
+   az provider register --namespace Microsoft.ExtendedLocation
+   ```
+
+   Monitor the registration process. Registration may take up to 10 minutes.
+
+   ```bash
+   az provider show -n Microsoft.Kubernetes -o table
+   az provider show -n Microsoft.KubernetesConfiguration -o table
+   az provider show -n Microsoft.ExtendedLocation -o table
+   ```
+
+   **NOTE**: This registration is needed only once per tenant.
+
+2. Create a resource group
+
+   Example:
+
+   ```bash
+   az group create --name AzureArcTest --location EastUS --output table
+   ```
+
+3. Register an application with the Microsoft identity platform
+
+   a. Register an application by following the steps outlined in the [Azure
+   documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#register-an-application).
+
+   b. Add credentials for the registered application by following the steps outlined
+[here](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#add-a-client-secret).
+
+      Take note of the Application ID, Tenant ID, Secret ID and Secret Value, in the overview section.
+      You'll need them in the component configuration.
+
+   c. Provide Azure Arc related permissions to the Service principal object(App registration). [Azure
+   documentation](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal)
+      provides a guided way to add permissions to the Service principal.
+
+   The following set of roles are needed
+     * Kubernetes Cluster - Azure Arc Onboarding
+     * Microsoft.Kubernetes connected cluster role
+
+## Configuration
+
+Azure Arc onboarding component configuration example:
+
+```tf
+# azure-arc-onboarding.lokocfg
+
+component azure-arc-onboarding {
```

Comment:
This is lacking double quotes.

Reply:
Its not needed actually but its good to add.